src/main/java/com/adyen/mirakl/service/util/ShopUtil.java

```
@@ -57,74 +57,80 @@ private ShopUtil() {
+            String postalCode = extractedKeysFromMirakl.getOrDefault(keys.get(POSTAL_CODE), null);
+            String stateOrProvince = extractedKeysFromMirakl.getOrDefault(keys.get(STATE_OR_PROVINCE), null);
+            String dateOfBirth = extractedKeysFromMirakl.getOrDefault(keys.get(DATE_OF_BIRTH), null);
+            String phoneNumber = extractedKeysFromMirakl.getOrDefault(keys.get(PHONE_NUMBER), null);
 
-            if (allMandatoryFieldsAvailable(firstName, lastName, civility, email)) {
+            //do nothing if mandatory fields are missing
+            if (firstName != null && lastName != null && civility != null && email != null) {
+
+                //mandatory fields
                 ShareholderContact shareholderContact = new ShareholderContact();
                 Name name = new Name();
                 name.setFirstName(firstName);
                 name.setLastName(lastName);
                 name.setGender(CIVILITY_TO_GENDER.getOrDefault(civility, Name.GenderEnum.UNKNOWN));
                 shareholderContact.setName(name);
                 shareholderContact.setEmail(email);
-                if(StringUtils.isNotEmpty(phoneNumber)){
+
+                //do not set Phone Number wrapper if the phone number does not exist
+                if (phoneNumber != null) {
                     final PhoneNumber phoneNumberWrapper = new PhoneNumber();
                     phoneNumberWrapper.setPhoneNumber(phoneNumber);
                     shareholderContact.setPhoneNumber(phoneNumberWrapper);
                 }
-                if(ImmutableList.of(country, street, houseNumberOrName, city, postalCode, stateOrProvince).stream().noneMatch(StringUtils::isBlank)){
+
+                //do not set the address unless at least one address attribute exists
+                if (country != null || street != null || houseNumberOrName != null || city != null || postalCode != null || stateOrProvince != null) {
```

minor: we could also do noneMatch -> !allMatch

---

same as above. otherwise all of this is just failing silently and nobody would no better.


dont log the details of the fields, just say address not set because missing fields...

---

Took out the noneMatch as I'm dealing with nulls now not empty strings. 
This makes it easier to not set a value if null e.g. `Optional.ofNullable(...).ifPresent(<setSomething>)`

---

Adyen seems to take empty strings as blanking the value and I get a bunch of validation errors back